File: bookOne/app/src/main/java/com/danny/bookone/MainActivity.java
package com.danny.bookone;

import android.os.Bundle;
import android.view.MenuItem;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.fragment.app.Fragment;
import androidx.viewpager.widget.ViewPager;

import com.danny.bookone.databinding.ActivityMainBinding;
import com.danny.bookone.fragment.MainPageViewPager;
import com.danny.bookone.fragment.Test2Fragment;
import com.danny.bookone.fragment.Test3Fragment;
import com.danny.bookone.fragment.BookFragment;
import com.danny.bookone.fragment.ThirdFrameFragment;
import com.google.android.material.bottomnavigation.BottomNavigationView;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding mainBinding;
    MainPageViewPager mainPageViewPager;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mainBinding = DataBindingUtil.setContentView(this, R.layout.activity_main);

        List<Fragment> fragments = new ArrayList<>();
        List<String> stringList = new ArrayList<>();

        stringList.add("one");
        stringList.add("two");
        stringList.add("three");
        stringList.add("four");

        fragments.add(BookFragment.newInstance());
        fragments.add(Test2Fragment.newInstance());
        fragments.add(Test3Fragment.newInstance());
        fragments.add(ThirdFrameFragment.newInstance());

        mainPageViewPager = new MainPageViewPager(getSupportFragmentManager(), fragments);
        mainPageViewPager.setTitle(stringList);
        mainBinding.viewPager.setAdapter(mainPageViewPager);
        mainBinding.viewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

            }

            @Override
            public void onPageSelected(int position) {
                mainBinding.bottomNav.getMenu().getItem(position).setChecked(true);
            }

            @Override
            public void onPageScrollStateChanged(int state) {
            }
        });
        mainBinding.bottomNav.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
                switch (menuItem.getItemId()) {
                    case R.id.item_bottom_1:
                        mainBinding.viewPager.setCurrentItem(0);
                        break;
                    case R.id.item_bottom_2:
                        mainBinding.viewPager.setCurrentItem(1);
                        break;
                    case R.id.item_bottom_3:
                        mainBinding.viewPager.setCurrentItem(2);
                        break;
                    case R.id.item_bottom_4:
                        mainBinding.viewPager.setCurrentItem(3);
                        break;
                }
                return false;
            }
        });
    }
}

File: bookOne/app/src/main/java/com/danny/bookone/MainActivity.kt
package com.danny.bookone

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import com.danny.bookone.Base.StringAdapter
import com.danny.bookone.aac.AacActivity
import com.danny.bookone.one.HeroActivity
import com.danny.bookone.other.CompanyActivity
import com.danny.bookone.other.DataActivity
import com.danny.bookone.other.HandlerActivity
import com.danny.bookone.highLight.HighLightActivity
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

class MainActivity : AppCompatActivity() {
    private val chapters = arrayOf(
            "Android群英传",
            "Android进阶之光",
            "AAC系列",
            "handler原理解析",
            "java数据结构",
            "Android四大组件")

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val stringList: List<String> = ArrayList(Arrays.asList(*chapters)) as List<String>
        val adapter = StringAdapter(stringList)
        rcv_chapter.layoutManager = LinearLayoutManager(this@MainActivity)
        rcv_chapter.addItemDecoration(DividerItemDecoration(this@MainActivity, DividerItemDecoration.VERTICAL))
        rcv_chapter.adapter = adapter
        adapter.setOnItemChildClickListener { adapter, view, position ->
            when (position) {
                0 -> startActivity(HeroActivity.createIntent(this@MainActivity))//Android群英传
                1 -> startActivity(HighLightActivity.createIntent(this@MainActivity))//android进阶之光
                2 -> startActivity(AacActivity.createIntent(this@MainActivity))//Jetpack架构篇
                3 -> startActivity(HandlerActivity.createIntent(this@MainActivity))//handler线程篇
                4 -> startActivity(DataActivity.createIntent(this@MainActivity))//数据结构篇。hashMap
                5 -> startActivity(CompanyActivity.createIntent(this@MainActivity))//android基础组件篇。服务，广播，activity
            }
        }
    }
}